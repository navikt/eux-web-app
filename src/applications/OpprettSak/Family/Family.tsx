import { FlexDiv, PileCenterDiv } from 'components/StyledComponents'
import { FamilieRelasjon, Kodeverk, Person } from 'declarations/types'
import _ from 'lodash'
import { Knapp } from 'nav-frontend-knapper'
import { Ingress, UndertekstBold, Undertittel } from 'nav-frontend-typografi'
import PersonCard from 'applications/OpprettSak/PersonCard/PersonCard'
import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'
import { HorizontalSeparatorDiv, VerticalSeparatorDiv } from 'nav-hoykontrast'
import AbroadPersonForm from './AbroadPersonForm'
import TPSPersonForm from './TPSPersonForm'

const FamilySubArea = styled.div`
  flex: 10;
`
const FamilySeparator = styled.div`
  flex: 1;
  border-left: 1px solid lightgrey;
`
const MarginDiv = styled.div`
  margin: 0.5rem;
`

export interface FamilyProps {
  abroadPersonFormAlertTypesWatched: Array<string> | undefined
  alertStatus: string | undefined
  alertMessage: string | undefined
  alertType: string | undefined
  familierelasjonKodeverk: Array<Kodeverk> | undefined
  onAbroadPersonAddedFailure: () => void
  onAbroadPersonAddedSuccess: (r: FamilieRelasjon) => void
  onAlertClose: () => void
  onRelationAdded: (p: Person | FamilieRelasjon) => void
  onRelationRemoved: (p: Person | FamilieRelasjon) => void
  onRelationReset: () => void
  onSearchFnr: (sok: any) => void
  onTPSPersonAddedFailure: () => void
  onTPSPersonAddedSuccess: (e: any) => void
  person: Person | undefined
  personRelatert: Person | undefined
  TPSPersonFormAlertTypesWatched: Array<string> | undefined
  valgteFamilieRelasjoner: Array<FamilieRelasjon> | undefined
}

const Family: React.FC<FamilyProps> = ({
  abroadPersonFormAlertTypesWatched,
  alertStatus,
  alertMessage,
  alertType,
  familierelasjonKodeverk,
  onAbroadPersonAddedFailure,
  onAbroadPersonAddedSuccess,
  onAlertClose,
  onRelationAdded,
  onRelationRemoved,
  onRelationReset,
  onSearchFnr,
  onTPSPersonAddedFailure,
  onTPSPersonAddedSuccess,
  personRelatert,
  person,
  valgteFamilieRelasjoner,
  TPSPersonFormAlertTypesWatched
}: FamilyProps): JSX.Element => {
  const [_viewAbroadPersonForm, setViewAbroadPersonForm] = useState<boolean>(false)
  const [_viewTPSRelatedForm, setViewTPSRelatedForm] = useState<boolean>(false)
  const { t } = useTranslation()

  const remainingRelationsFromTPS: Array<FamilieRelasjon> = _.filter(person!.relasjoner, (relation: FamilieRelasjon) =>
    _.find(valgteFamilieRelasjoner, (valgteRelasjon: FamilieRelasjon) => valgteRelasjon.fnr === relation.fnr) === undefined
  )

  const toggleViewAbroadPersonForm = (): void => {
    setViewAbroadPersonForm(!_viewAbroadPersonForm)
  }

  const toggleViewTPSRelatedForm = (): void => {
    setViewTPSRelatedForm(!_viewTPSRelatedForm)
  }

  const ekskluderteVerdier: Array<string> = []

  if (!_.isEmpty(valgteFamilieRelasjoner)) {
    // Hvis ektefelle allerede er lagt til, fjern mulighet for andre typer samlivspartnere
    if (valgteFamilieRelasjoner?.find((relation: FamilieRelasjon) => relation.rolle === 'EKTE')) {
      ekskluderteVerdier.push('EKTE', 'SAMB', 'REPA')
    }
    // Hvis registret partner allerede er lagt til, fjern mulighet for andre typer samlivspartnere
    if (valgteFamilieRelasjoner?.find((relation: FamilieRelasjon) => relation.rolle === 'REPA')) {
      ekskluderteVerdier.push('EKTE', 'SAMB', 'REPA')
    }
    // Det skal kun være mulig å legge til en relasjon av typen annen
    if (valgteFamilieRelasjoner?.find((relation: FamilieRelasjon) => relation.rolle === 'ANNEN')) {
      ekskluderteVerdier.push('ANNEN')
    }
  }

  const rolleList: Array<Kodeverk> = familierelasjonKodeverk!.filter(
    (kt: Kodeverk) => ekskluderteVerdier.includes(kt.kode) === false
  )

  return (
    <div data-test-id='c-family'>
      <HorizontalSeparatorDiv />
      <Undertittel>
        {t('el:title-family-description')}
      </Undertittel>
      <VerticalSeparatorDiv />
      <FlexDiv>
        <FamilySubArea>
          <HorizontalSeparatorDiv />
          <Ingress>
            {t('label:family-relations-in-tps')}
          </Ingress>
          {remainingRelationsFromTPS.map((relation: FamilieRelasjon) => (
            <MarginDiv key={relation.fnr}>
              <PersonCard
                className='slideInFromLeft personNotSelected'
                familierelasjonKodeverk={familierelasjonKodeverk}
                key={relation.fnr}
                onAddClick={onRelationAdded}
                person={relation}
              />
            </MarginDiv>
          ))}
          {!_.isEmpty(person!.relasjoner) && _.isEmpty(remainingRelationsFromTPS) && (
            <>
              <HorizontalSeparatorDiv data-size='0.5' />
              <VerticalSeparatorDiv data-size='1.5' />
              <UndertekstBold>
                ({t('label:family-added-all')})
              </UndertekstBold>
            </>
          )}
          {_.isEmpty(person!.relasjoner) && (
            <>
              <HorizontalSeparatorDiv data-size='0.5' />
              <VerticalSeparatorDiv data-size='1.5' />
              <UndertekstBold>
                ({t('label:family-none-in-tps')})
              </UndertekstBold>
            </>
          )}
        </FamilySubArea>
        <FamilySeparator />
        <FamilySubArea>
          <HorizontalSeparatorDiv />
          <Ingress>
            {t('label:family-chosen')}&nbsp;({valgteFamilieRelasjoner ? valgteFamilieRelasjoner.length : 0})
          </Ingress>
          {valgteFamilieRelasjoner && valgteFamilieRelasjoner.map((relation: FamilieRelasjon) => (
            <MarginDiv key={relation.fnr}>
              <PersonCard
                className='slideInFromLeft personSelected'
                familierelasjonKodeverk={familierelasjonKodeverk}
                key={relation.fnr}
                onRemoveClick={onRelationRemoved}
                person={relation}
              />
            </MarginDiv>
          ))}
        </FamilySubArea>
      </FlexDiv>
      <PileCenterDiv>
        <div>
          <VerticalSeparatorDiv data-size='1.5' />
          <Ingress>
            {t('el:title-family-utland')}
          </Ingress>
          {_viewAbroadPersonForm && (
            <>
              <VerticalSeparatorDiv />
              <AbroadPersonForm
                alertStatus={alertStatus}
                alertMessage={alertMessage}
                alertType={alertType}
                alertTypesWatched={abroadPersonFormAlertTypesWatched}
                existingFamilyRelationships={(valgteFamilieRelasjoner || []).concat(remainingRelationsFromTPS || [])}
                onAbroadPersonAddedFailure={onAbroadPersonAddedFailure}
                onAbroadPersonAddedSuccess={onAbroadPersonAddedSuccess}
                onAlertClose={onAlertClose}
                person={person}
                rolleList={rolleList}
              />
            </>
          )}
          <VerticalSeparatorDiv />
          <Knapp
            onClick={toggleViewAbroadPersonForm}
          >
            {_viewAbroadPersonForm
              ? t('label:hide-form')
              : t('label:show-form')}
          </Knapp>
        </div>
        <div>
          <VerticalSeparatorDiv data-size='1.5' />
          <Ingress>
            {t('el:title-family-tps')}
          </Ingress>
          {_viewTPSRelatedForm && person && (
            <>
              <VerticalSeparatorDiv />
              <TPSPersonForm
                alertStatus={alertStatus}
                alertMessage={alertMessage}
                alertType={alertType}
                alertTypesWatched={TPSPersonFormAlertTypesWatched}
                existingFamilyRelationships={(valgteFamilieRelasjoner || []).concat(remainingRelationsFromTPS || [])}
                onAlertClose={onAlertClose}
                onSearchFnr={onSearchFnr}
                onRelationReset={onRelationReset}
                onTPSPersonAddedFailure={onTPSPersonAddedFailure}
                onTPSPersonAddedSuccess={onTPSPersonAddedSuccess}
                person={person}
                personRelatert={personRelatert}
                rolleList={rolleList}
              />
            </>
          )}
          <VerticalSeparatorDiv />
          <Knapp onClick={toggleViewTPSRelatedForm}>
            {_viewTPSRelatedForm
              ? t('label:hide-form')
              : t('label:show-form')}
          </Knapp>
        </div>
      </PileCenterDiv>
    </div>
  )
}

export default Family
