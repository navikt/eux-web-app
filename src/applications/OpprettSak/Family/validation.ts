import { OldFamilieRelasjon, Person, Validation } from 'declarations/types'
import { addError, checkIfNotEmpty } from 'utils/validation'

export interface AbroadPersonFormValidationProps {
  relation: OldFamilieRelasjon,
  namespace: string
}

export const validateAbroadPersonForm = (
  v: Validation,
  namespace: string,
  {
    relation
  }: AbroadPersonFormValidationProps
): boolean => {
  const hasErrors: Array<boolean> = []

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.fnr,
    id: namespace + '-fnr',
    message: 'validation:noFnr'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.fornavn,
    id: namespace + '-fornavn',
    message: 'validation:noFirstName'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.etternavn,
    id: namespace + '-etternavn',
    message: 'validation:noLastName'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.kjoenn,
    id: namespace + '-kjoenn',
    message: 'validation:noGender'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.fdato,
    id: namespace + '-fdato',
    message: 'validation:noDate'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.rolle,
    id: namespace + '-familierelasjon',
    message: 'validation:noRolle'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.land,
    id: namespace + '-land',
    message: 'validation:noLand'
  }))

  hasErrors.push(checkIfNotEmpty(v, {
    needle: relation.statsborgerskap,
    id: namespace + '-statsborgerskap',
    message: 'validation:noNationality'
  }))

  return hasErrors.find(value => value) !== undefined
}

export interface TPSPersonFormValidationProps {
  fnr: string,
  tpsperson: OldFamilieRelasjon | undefined
  person: Person
}

export const validateTPSPersonForm = (
  v: Validation,
  namespace: string,
  {
    person,
    fnr
  }: TPSPersonFormValidationProps
): boolean => {
  const hasErrors: Array<boolean> = []

  hasErrors.push(checkIfNotEmpty(v, {
    needle: fnr,
    id: namespace + '-fnr-dnr',
    message: 'validation:noFnr'
  }))

  if (person.fnr === fnr) {
    hasErrors.push(true)
    addError(v, {
      id: namespace + '-fnr-dnr',
      message: 'message:error-fnr-is-user',
      extra: { sok: fnr }
    })
  }

  return hasErrors.find(value => value) !== undefined
}
