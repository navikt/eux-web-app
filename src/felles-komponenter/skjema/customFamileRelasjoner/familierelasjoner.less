@import '../../../constants';

.familerelasjoner {

  .familierelasjoner__linje {
    display: flex;
    flex-direction: row;
    margin: 2em 0;
    width: 800px;
  }

  .familierelasjoner__input {
    margin: 0 1em 0 0;
  }

  .familierelasjoner__detailjer {
    display: flex;
    width: 40%;
    flex-direction: column;
    margin: 0;
  }

  .familierelasjoner__tittel, .familierelasjoner__detalj {
    margin: 0;
    padding: 0;
  }

  .familierelasjoner__tittel {
    font-weight: bold;
  }

  .familierelasjoner__knapp {
    display: flex;
    align-self: flex-start;
    margin-top: 1.9em;
    align-items: center;
  }

  .familierelasjoner__knapp__ikon {
    margin-right: .5em;

    path {
      stroke: @blue-nav;
    }
  }

  .familierelasjoner__knapp {
    &:hover {
      .familierelasjoner__knapp__ikon path {
        stroke: @white !important;
      }
    }
  }
}
