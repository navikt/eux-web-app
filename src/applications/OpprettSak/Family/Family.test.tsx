import { familierelasjoner } from 'eessi-kodeverk/lib/familierelasjoner'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import samplePerson from 'resources/tests/samplePerson'
import Family, { FamilyProps } from './Family'

describe('components/Family/Family', () => {
  let wrapper: ReactWrapper

  const initialMockProps: FamilyProps = {
    alertStatus: undefined,
    alertMessage: undefined,
    alertType: undefined,
    familierelasjonKodeverk: familierelasjoner,
    onAbroadPersonAddedFailure: jest.fn(),
    onAbroadPersonAddedSuccess: jest.fn(),
    onRelationAdded: jest.fn(),
    onRelationRemoved: jest.fn(),
    onRelationReset: jest.fn(),
    onSearchFnr: jest.fn(),
    onTPSPersonAddedFailure: jest.fn(),
    onTPSPersonAddedSuccess: jest.fn(),
    personRelatert: undefined,
    person: samplePerson,
    valgteFamilieRelasjoner: undefined,
    abroadPersonFormAlertTypesWatched: [],
    TPSPersonFormAlertTypesWatched: []
  }

  beforeEach(() => {
    wrapper = mount(<Family {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('HTML structure', () => {
    expect(wrapper.exists('.c-family')).toBeTruthy()
  })
})
