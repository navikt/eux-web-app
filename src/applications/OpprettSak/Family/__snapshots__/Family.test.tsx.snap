// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Family/Family Renders 1`] = `
.c0 {
  display: inline-block;
  margin-left: 1rem;
}

.c1 {
  margin-bottom: 1rem;
}

.c13 {
  margin-bottom: 1.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  border: 1px solid lightgray;
  background: transparent;
  margin-top: 1.5rem;
  max-width: 800px;
  min-width: 400px;
  padding: 1rem;
  border-radius: 5px;
}

.c5.personNotSelected {
  border: 3px solid red;
}

.c5.personSelected {
  border: 3px solid green;
}

.c5.neutral {
  background: white;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 {
  -webkit-flex: 10;
  -ms-flex: 10;
  flex: 10;
}

.c10 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-left: 1px solid lightgrey;
}

.c4 {
  margin: 0.5rem;
}

<Family
  familierelasjonKodeverk={
    Array [
      Object {
        "kode": "BARN",
        "term": "Barn av",
      },
      Object {
        "kode": "EKTE",
        "term": "Ektefelle til",
      },
      Object {
        "kode": "REPA",
        "term": "Registrert partner med",
      },
      Object {
        "kode": "SAMB",
        "term": "Samboer med",
      },
      Object {
        "kode": "ANNEN",
        "term": "Annen person",
      },
    ]
  }
  onAbroadPersonAddedFailure={[MockFunction]}
  onAbroadPersonAddedSuccess={[MockFunction]}
  onAlertClose={[MockFunction]}
  onRelationAdded={[MockFunction]}
  onRelationRemoved={[MockFunction]}
  onRelationReset={[MockFunction]}
  onSearchFnr={[MockFunction]}
  onTPSPersonAddedFailure={[MockFunction]}
  onTPSPersonAddedSuccess={[MockFunction]}
  person={
    Object {
      "etternavn": "BLYANT",
      "fdato": "1961-02-02",
      "fnr": "02026100715",
      "fornavn": "STOR",
      "kjoenn": "M",
      "relasjoner": Array [
        Object {
          "etternavn": "BLYANT",
          "fdato": "1961-04-08",
          "fnr": "02026100714",
          "fornavn": "PEN",
          "kjoenn": "K",
          "rolle": "EKTE",
        },
        Object {
          "etternavn": "BLYANT",
          "fdato": "1980-02-02",
          "fnr": "02026100713",
          "fornavn": "LITEN",
          "kjoenn": "M",
          "rolle": "BARN",
        },
      ],
    }
  }
>
  <div
    data-test-id="c-family"
  >
    <styled.div>
      <div
        className="c0"
      />
    </styled.div>
    <Undertittel>
      <TypografiBase
        type="undertittel"
      >
        <h2
          className="typo-undertittel"
        >
          el:title-family-description
        </h2>
      </TypografiBase>
    </Undertittel>
    <styled.div>
      <div
        className="c1"
      />
    </styled.div>
    <styled.div>
      <div
        className="c2"
      >
        <styled.div>
          <div
            className="c3"
          >
            <styled.div>
              <div
                className="c0"
              />
            </styled.div>
            <Ingress>
              <TypografiBase
                type="ingress"
              >
                <p
                  className="typo-ingress"
                >
                  label:family-relations-in-tps
                </p>
              </TypografiBase>
            </Ingress>
            <styled.div
              key="02026100714"
            >
              <div
                className="c4"
              >
                <PersonCard
                  className="slideInFromLeft personNotSelected"
                  familierelasjonKodeverk={
                    Array [
                      Object {
                        "kode": "BARN",
                        "term": "Barn av",
                      },
                      Object {
                        "kode": "EKTE",
                        "term": "Ektefelle til",
                      },
                      Object {
                        "kode": "REPA",
                        "term": "Registrert partner med",
                      },
                      Object {
                        "kode": "SAMB",
                        "term": "Samboer med",
                      },
                      Object {
                        "kode": "ANNEN",
                        "term": "Annen person",
                      },
                    ]
                  }
                  key="02026100714"
                  onAddClick={[MockFunction]}
                  person={
                    Object {
                      "etternavn": "BLYANT",
                      "fdato": "1961-04-08",
                      "fnr": "02026100714",
                      "fornavn": "PEN",
                      "kjoenn": "K",
                      "rolle": "EKTE",
                    }
                  }
                >
                  <Styled(Panel)
                    className="slideInFromLeft personNotSelected"
                  >
                    <Panel
                      border={false}
                      className="c5 slideInFromLeft personNotSelected"
                    >
                      <div
                        className="panel c5 slideInFromLeft personNotSelected"
                      >
                        <styled.div>
                          <div
                            className="c6"
                          >
                            <styled.div>
                              <div
                                className="c7"
                              >
                                <img
                                  alt="nav-woman-icon"
                                  height={50}
                                  src="Woman.png"
                                  width={50}
                                />
                                <styled.div>
                                  <div
                                    className="c0"
                                  />
                                </styled.div>
                                <div
                                  data-test-id="panelheader__tittel"
                                >
                                  <Undertittel
                                    data-test-id="panelheader__tittel__hoved"
                                  >
                                    <TypografiBase
                                      data-test-id="panelheader__tittel__hoved"
                                      type="undertittel"
                                    >
                                      <h2
                                        className="typo-undertittel"
                                        data-test-id="panelheader__tittel__hoved"
                                      >
                                        PEN
                                         
                                        BLYANT
                                         - Ektefelle til
                                      </h2>
                                    </TypografiBase>
                                  </Undertittel>
                                  <styled.div>
                                    <div
                                      className="c8"
                                    >
                                      <div>
                                        label:fnr : 02026100714
                                      </div>
                                      <div>
                                        label:birthdate: 08.04.1961
                                      </div>
                                    </div>
                                  </styled.div>
                                </div>
                              </div>
                            </styled.div>
                            <Knapp
                              data-test-id="familierelasjoner__knapp--legg-til"
                              disabled={false}
                              onClick={[Function]}
                            >
                              <KnappBase
                                autoDisableVedSpinner={false}
                                className={null}
                                data-test-id="familierelasjoner__knapp--legg-til"
                                disabled={false}
                                htmlType="submit"
                                inaktivKlasseVedDisabled={false}
                                kompakt={false}
                                mini={false}
                                onClick={[Function]}
                                spinner={false}
                                type="standard"
                              >
                                <button
                                  className="knapp knapp--standard"
                                  data-test-id="familierelasjoner__knapp--legg-til"
                                  disabled={false}
                                  onClick={[Function]}
                                  type="submit"
                                >
                                  <_default
                                    width={20}
                                  >
                                    <svg
                                      version="1.1"
                                      viewBox="0 0 24 24"
                                      width={20}
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <title>
                                        Add
                                      </title>
                                      <g
                                        fill="none"
                                        fillRule="evenodd"
                                        stroke="none"
                                        strokeWidth="1"
                                      >
                                        <path
                                          d="M20.146852,3.88403947 L20.146771,3.88395809 C17.976089,1.70435546 15.0928534,0.505997541 11.9994782,0.5 C5.67074719,0.5 0.512021923,5.64860101 0.500022608,11.9769766 C0.494020029,15.0501691 1.68354863,17.9384927 3.85210448,20.1159605 C6.02125562,22.2930255 8.90491868,23.494 11.9775643,23.5 C18.3251119,23.5 23.4879722,18.3534487 23.4999774,12.0209361 C23.5059792,8.94917357 22.3156837,6.06078388 20.146852,3.88403947 Z"
                                          stroke="#0067C5"
                                        />
                                        <path
                                          d="M18.2616513,12.5217391 L5.73939227,12.5217391 C5.45138031,12.5206957 5.21658796,12.288 5.21763148,12 C5.21763148,11.7130435 5.45138031,11.4782609 5.73939227,11.4782609 L18.2616513,11.4782609 C18.5486197,11.4782609 18.7844556,11.712 18.7834121,12 C18.7834121,12.288 18.5496632,12.5217391 18.2616513,12.5217391 Z"
                                          fill="#0067C5"
                                        />
                                        <path
                                          d="M18.2616513,12.5217391 L5.73939227,12.5217391 C5.45138031,12.5206957 5.21658796,12.288 5.21763148,12 C5.21763148,11.7130435 5.45138031,11.4782609 5.73939227,11.4782609 L18.2616513,11.4782609 C18.5486197,11.4782609 18.7844556,11.712 18.7834121,12 C18.7834121,12.288 18.5496632,12.5217391 18.2616513,12.5217391 Z"
                                          fill="#0067C5"
                                          transform="translate(12.000522, 12.000000) rotate(90.000000) translate(-12.000522, -12.000000) "
                                        />
                                      </g>
                                    </svg>
                                  </_default>
                                  <styled.div>
                                    <div
                                      className="c0"
                                    />
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="c9"
                                    >
                                      label:add
                                    </div>
                                  </styled.div>
                                </button>
                              </KnappBase>
                            </Knapp>
                          </div>
                        </styled.div>
                      </div>
                    </Panel>
                  </Styled(Panel)>
                </PersonCard>
              </div>
            </styled.div>
            <styled.div
              key="02026100713"
            >
              <div
                className="c4"
              >
                <PersonCard
                  className="slideInFromLeft personNotSelected"
                  familierelasjonKodeverk={
                    Array [
                      Object {
                        "kode": "BARN",
                        "term": "Barn av",
                      },
                      Object {
                        "kode": "EKTE",
                        "term": "Ektefelle til",
                      },
                      Object {
                        "kode": "REPA",
                        "term": "Registrert partner med",
                      },
                      Object {
                        "kode": "SAMB",
                        "term": "Samboer med",
                      },
                      Object {
                        "kode": "ANNEN",
                        "term": "Annen person",
                      },
                    ]
                  }
                  key="02026100713"
                  onAddClick={[MockFunction]}
                  person={
                    Object {
                      "etternavn": "BLYANT",
                      "fdato": "1980-02-02",
                      "fnr": "02026100713",
                      "fornavn": "LITEN",
                      "kjoenn": "M",
                      "rolle": "BARN",
                    }
                  }
                >
                  <Styled(Panel)
                    className="slideInFromLeft personNotSelected"
                  >
                    <Panel
                      border={false}
                      className="c5 slideInFromLeft personNotSelected"
                    >
                      <div
                        className="panel c5 slideInFromLeft personNotSelected"
                      >
                        <styled.div>
                          <div
                            className="c6"
                          >
                            <styled.div>
                              <div
                                className="c7"
                              >
                                <img
                                  alt="nav-man-icon"
                                  height={50}
                                  src="Man.png"
                                  width={50}
                                />
                                <styled.div>
                                  <div
                                    className="c0"
                                  />
                                </styled.div>
                                <div
                                  data-test-id="panelheader__tittel"
                                >
                                  <Undertittel
                                    data-test-id="panelheader__tittel__hoved"
                                  >
                                    <TypografiBase
                                      data-test-id="panelheader__tittel__hoved"
                                      type="undertittel"
                                    >
                                      <h2
                                        className="typo-undertittel"
                                        data-test-id="panelheader__tittel__hoved"
                                      >
                                        LITEN
                                         
                                        BLYANT
                                         - Barn av
                                      </h2>
                                    </TypografiBase>
                                  </Undertittel>
                                  <styled.div>
                                    <div
                                      className="c8"
                                    >
                                      <div>
                                        label:fnr : 02026100713
                                      </div>
                                      <div>
                                        label:birthdate: 02.02.1980
                                      </div>
                                    </div>
                                  </styled.div>
                                </div>
                              </div>
                            </styled.div>
                            <Knapp
                              data-test-id="familierelasjoner__knapp--legg-til"
                              disabled={false}
                              onClick={[Function]}
                            >
                              <KnappBase
                                autoDisableVedSpinner={false}
                                className={null}
                                data-test-id="familierelasjoner__knapp--legg-til"
                                disabled={false}
                                htmlType="submit"
                                inaktivKlasseVedDisabled={false}
                                kompakt={false}
                                mini={false}
                                onClick={[Function]}
                                spinner={false}
                                type="standard"
                              >
                                <button
                                  className="knapp knapp--standard"
                                  data-test-id="familierelasjoner__knapp--legg-til"
                                  disabled={false}
                                  onClick={[Function]}
                                  type="submit"
                                >
                                  <_default
                                    width={20}
                                  >
                                    <svg
                                      version="1.1"
                                      viewBox="0 0 24 24"
                                      width={20}
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <title>
                                        Add
                                      </title>
                                      <g
                                        fill="none"
                                        fillRule="evenodd"
                                        stroke="none"
                                        strokeWidth="1"
                                      >
                                        <path
                                          d="M20.146852,3.88403947 L20.146771,3.88395809 C17.976089,1.70435546 15.0928534,0.505997541 11.9994782,0.5 C5.67074719,0.5 0.512021923,5.64860101 0.500022608,11.9769766 C0.494020029,15.0501691 1.68354863,17.9384927 3.85210448,20.1159605 C6.02125562,22.2930255 8.90491868,23.494 11.9775643,23.5 C18.3251119,23.5 23.4879722,18.3534487 23.4999774,12.0209361 C23.5059792,8.94917357 22.3156837,6.06078388 20.146852,3.88403947 Z"
                                          stroke="#0067C5"
                                        />
                                        <path
                                          d="M18.2616513,12.5217391 L5.73939227,12.5217391 C5.45138031,12.5206957 5.21658796,12.288 5.21763148,12 C5.21763148,11.7130435 5.45138031,11.4782609 5.73939227,11.4782609 L18.2616513,11.4782609 C18.5486197,11.4782609 18.7844556,11.712 18.7834121,12 C18.7834121,12.288 18.5496632,12.5217391 18.2616513,12.5217391 Z"
                                          fill="#0067C5"
                                        />
                                        <path
                                          d="M18.2616513,12.5217391 L5.73939227,12.5217391 C5.45138031,12.5206957 5.21658796,12.288 5.21763148,12 C5.21763148,11.7130435 5.45138031,11.4782609 5.73939227,11.4782609 L18.2616513,11.4782609 C18.5486197,11.4782609 18.7844556,11.712 18.7834121,12 C18.7834121,12.288 18.5496632,12.5217391 18.2616513,12.5217391 Z"
                                          fill="#0067C5"
                                          transform="translate(12.000522, 12.000000) rotate(90.000000) translate(-12.000522, -12.000000) "
                                        />
                                      </g>
                                    </svg>
                                  </_default>
                                  <styled.div>
                                    <div
                                      className="c0"
                                    />
                                  </styled.div>
                                  <styled.div>
                                    <div
                                      className="c9"
                                    >
                                      label:add
                                    </div>
                                  </styled.div>
                                </button>
                              </KnappBase>
                            </Knapp>
                          </div>
                        </styled.div>
                      </div>
                    </Panel>
                  </Styled(Panel)>
                </PersonCard>
              </div>
            </styled.div>
          </div>
        </styled.div>
        <styled.div>
          <div
            className="c10"
          />
        </styled.div>
        <styled.div>
          <div
            className="c3"
          >
            <styled.div>
              <div
                className="c0"
              />
            </styled.div>
            <Ingress>
              <TypografiBase
                type="ingress"
              >
                <p
                  className="typo-ingress"
                >
                  label:family-chosen
                   (
                  0
                  )
                </p>
              </TypografiBase>
            </Ingress>
          </div>
        </styled.div>
      </div>
    </styled.div>
    <Styled(styled.div)>
      <div
        className="c11 c12"
      >
        <div>
          <styled.div
            data-size="1.5"
          >
            <div
              className="c13"
              data-size="1.5"
            />
          </styled.div>
          <Ingress>
            <TypografiBase
              type="ingress"
            >
              <p
                className="typo-ingress"
              >
                el:title-family-utland
              </p>
            </TypografiBase>
          </Ingress>
          <styled.div>
            <div
              className="c1"
            />
          </styled.div>
          <Knapp
            onClick={[Function]}
          >
            <KnappBase
              autoDisableVedSpinner={false}
              className={null}
              disabled={false}
              htmlType="submit"
              inaktivKlasseVedDisabled={false}
              kompakt={false}
              mini={false}
              onClick={[Function]}
              spinner={false}
              type="standard"
            >
              <button
                className="knapp knapp--standard"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                label:show-form
              </button>
            </KnappBase>
          </Knapp>
        </div>
        <div>
          <styled.div
            data-size="1.5"
          >
            <div
              className="c13"
              data-size="1.5"
            />
          </styled.div>
          <Ingress>
            <TypografiBase
              type="ingress"
            >
              <p
                className="typo-ingress"
              >
                el:title-family-tps
              </p>
            </TypografiBase>
          </Ingress>
          <styled.div>
            <div
              className="c1"
            />
          </styled.div>
          <Knapp
            onClick={[Function]}
          >
            <KnappBase
              autoDisableVedSpinner={false}
              className={null}
              disabled={false}
              htmlType="submit"
              inaktivKlasseVedDisabled={false}
              kompakt={false}
              mini={false}
              onClick={[Function]}
              spinner={false}
              type="standard"
            >
              <button
                className="knapp knapp--standard"
                disabled={false}
                onClick={[Function]}
                type="submit"
              >
                label:show-form
              </button>
            </KnappBase>
          </Knapp>
        </div>
      </div>
    </Styled(styled.div)>
  </div>
</Family>
`;
